ds248x/ds248x.go

```diff
@@ -154,6 +170,65 @@ func (d *Dev) Search(alarmOnly bool) ([]onewire.Address, error) {
 	return onewire.Search(d, alarmOnly)
 }
 
+// ChannelSelect function is for selecting one of eight 1-w channels on DS2482-800.
+// On other chips it does nothing. Channel selection values are between
+// 0 and 7. It is expected that application keeps track of
+// with 1-w device is connected to with channel.
+// Communication error is returned if present.
+func (d *Dev) ChannelSelect(ch int) error {
+	var err error = nil
+	switch d.isDS248x {
+	case isDS2482x100:
+		// err = errors.New("unsupported (ds2482-100 has only channel 0)")
+		// return fmt.Errorf("ds2482-100: error while selecting channel: %s", err)
+	case isDS2482x800:
+		if ch < 0 || ch > 7 {
+			err = errors.New("channel out of range 0...7")
+			return fmt.Errorf("ds2482-800: error while selecting channel: %s", err)
+		}
+		buf := []byte{cmdChannelSelect, cscw[ch]}
+		if err = d.i2c.Tx(buf, nil); err != nil {
+			return fmt.Errorf("ds2482-800: error while selecting channel: %s", err)
+		}
+	case isDS2483:
+		// err = errors.New("unsupported (ds2483 has only channel 0)")
+		// return fmt.Errorf("ds2483: error while selecting channel: %s", err)
+	default:
+		err = errors.New("wrong chip")
+		return fmt.Errorf("ds248x: error while selecting channel: %s", err)
+	}
+	return err
```

Comment:
This will not be needed once you convert all the case to have a return statement.

Reply:
Some environments complains about this.

Reply:
It's because you don't return properly in the switch clauses.